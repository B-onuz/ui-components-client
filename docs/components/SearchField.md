# SearchField (Component)

O componente de campo de busca recebe as mesmas props do component input

[Documentação](Input.md)

## Como utilizar

```js
import SearchField from 'ui-components/lib/components/SearchField'
```

```jsx
<SearchField />
```
