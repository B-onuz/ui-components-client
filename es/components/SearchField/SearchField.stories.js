'use strict'

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.SearchFieldExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _SearchField = _interopRequireDefault(require('./SearchField'))

var _reflexbox = require('reflexbox')

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/SearchField/SearchField.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'SearchField',
  component: _SearchField['default'],
}
exports['default'] = _default

var SearchFieldExample = function SearchFieldExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _reflexbox.Flex,
    {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 11,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _reflexbox.Box,
      {
        width: [1 / 2],
        m: 20,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 12,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(_SearchField['default'], {
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 13,
          columnNumber: 7,
        },
      })
    )
  )
}

exports.SearchFieldExample = SearchFieldExample
