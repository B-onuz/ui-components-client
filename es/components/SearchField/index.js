"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _SearchField = _interopRequireDefault(require("./SearchField"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _SearchField["default"];
exports["default"] = _default;