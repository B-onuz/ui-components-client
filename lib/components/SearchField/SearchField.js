"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _Input = _interopRequireDefault(require("../Input"));

var _reactFontawesome = require("@fortawesome/react-fontawesome");

var _freeSolidSvgIcons = require("@fortawesome/free-solid-svg-icons");

var _theme = _interopRequireDefault(require("../../theme/"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/SearchField/SearchField.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  right: 12px;\n  color: ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  display: flex;\n  background-color: #fff;\n  justify-content: center;\n  align-items: center;\n  position: relative;\n  border-radius: 6px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  flex: 1;\n  background-color: transparent;\n  padding-right: 32px;\n  &:focus {\n    background-color: #fff;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledSearchField = (0, _styledComponents["default"])(_Input["default"])(_templateObject());

var Wrapper = _styledComponents["default"].div(_templateObject2(), (0, _styledSystem.compose)(_styledSystem.space, _styledSystem.layout));

var Icon = (0, _styledComponents["default"])(_reactFontawesome.FontAwesomeIcon)(_templateObject3(), function (_ref) {
  var theme = _ref.theme;
  return theme.colors.darkGrey;
});

var SearchField = function SearchField(_ref2) {
  var children = _ref2.children,
      rest = _objectWithoutPropertiesLoose(_ref2, ["children"]);

  return /*#__PURE__*/_react["default"].createElement(Wrapper, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 37,
      columnNumber: 5
    }
  }, /*#__PURE__*/_react["default"].createElement(StyledSearchField, _extends({
    placeholder: "Pesquisar"
  }, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 38,
      columnNumber: 7
    }
  })), /*#__PURE__*/_react["default"].createElement(Icon, {
    icon: _freeSolidSvgIcons.faSearch,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 39,
      columnNumber: 7
    }
  }));
};

var _default = SearchField;
exports["default"] = _default;
module.exports = exports.default;