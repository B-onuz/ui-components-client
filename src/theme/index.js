const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
