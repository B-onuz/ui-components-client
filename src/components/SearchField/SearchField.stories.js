import React from 'react'
import SearchField from './SearchField'
import { Flex, Box } from 'reflexbox'

export default {
  title: 'SearchField',
  component: SearchField,
}

export const SearchFieldExample = () => (
  <Flex>
    <Box width={[1 / 2]} m={20}>
      <SearchField />
    </Box>
  </Flex>
)
