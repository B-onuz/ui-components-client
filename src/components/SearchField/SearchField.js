import React from 'react'
import styled from 'styled-components'
import { color } from 'styled-system'
import Input from '../Input'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSearch } from '@fortawesome/free-solid-svg-icons'
import { space, compose, layout } from 'styled-system'
import theme from '../../theme/'

const StyledSearchField = styled(Input)`
  flex: 1;
  background-color: transparent;
  padding-right: 32px;
  &:focus {
    background-color: #fff;
  }
`

const Wrapper = styled.div`
  ${compose(space, layout)}
  display: flex;
  background-color: #fff;
  justify-content: center;
  align-items: center;
  position: relative;
  border-radius: 6px;
`

const Icon = styled(FontAwesomeIcon)`
  position: absolute;
  right: 12px;
  color: ${({ theme }) => theme.colors.darkGrey};
`

const SearchField = ({ children, ...rest }) => {
  return (
    <Wrapper>
      <StyledSearchField placeholder="Pesquisar" {...rest} />
      <Icon icon={faSearch} />
    </Wrapper>
  )
}

export default SearchField
