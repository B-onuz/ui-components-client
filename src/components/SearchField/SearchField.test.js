import React from 'react'
import SearchField from './SearchField.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render SearchField component', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <SearchField id="my-SearchField" placeholder="My SearchField" />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
