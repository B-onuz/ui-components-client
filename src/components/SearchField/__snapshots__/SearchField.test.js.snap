// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SearchField component 1`] = `
.c1 {
  border: 1px solid #dededf;
  border-radius: 4px;
  line-height: 2.1em;
  outline: none;
  padding-right: 1em;
  padding-left: 1em;
  width: 100%;
  box-sizing: border-box;
  font-size: 16px;
}

.c1::-webkit-input-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c1::-moz-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c1:-ms-input-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c1::placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c1:focus {
  box-shadow: 0 0 3px rgba(33,133,208,0.5);
  border-color: #2185d0;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background-color: transparent;
  padding-right: 32px;
}

.c2:focus {
  background-color: #fff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-radius: 6px;
}

.c3 {
  position: absolute;
  right: 12px;
  color: #9A9A9A;
}

<div
  className="c0"
>
  <input
    className="c1 c2"
    color="blue"
    id="my-SearchField"
    placeholder="My SearchField"
    size="medium"
  />
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-search fa-w-16 c3"
    data-icon="search"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;
